doc/man/man3/XMP.3.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) doc/man/man3/XMP.3.man
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2011  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual under
.\" the terms of the GNU Free Documentation License, Version 1.3 or any later
.\" version published by the Free Software Foundation; with no Invariant
.\" Sections, no Front-Cover Texts, and no Back-Cover Texts.  A copy of the
.\" license is included in the section entitled "GNU Free Documentation
.\" License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaining to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF SUCH
.\" CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this manual
.\" page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from the
.\" use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
.\" behalf of the U.S. Government ("Government"), the following provisions
.\" apply to you.  If the Software is supplied by the Department of Defense
.\" ("DoD"), it is classified as "Commercial Computer Software" under
.\" paragraph 252.227-7014 of the DoD Supplement to the Federal Acquisition
.\" Regulations ("DFARS") (or any successor regulations) and the Government
.\" is acquiring only the license rights granted herein (the license rights
.\" customarily provided to non-Government users).  If the Software is
.\" supplied to any unit or agency of the Government other than DoD, it is
.\" classified as "Restricted Computer Software" and the Government's rights
.\" in the Software are defined in paragraph 52.227-19 of the Federal
.\" Acquisition Regulations ("FAR") (or any successor regulations) or, in the
.\" cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the
.\" FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore CGIQOSTU
.R2
.so openss7.macros
.\"
.\"
.TH XMP 3 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Library Calls"
.\"
.\"
.SH NAME
.B libxmp, xmp, XMP
\- X/Open Management Protocol API (XMP)
.\"
.\"
.SH SYNOPSIS
.PP
.HP
\fBMP_status mp_abandon(OM_private_object \fIsession\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_abort_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB);\fR
.HP
\fBMP_status mp_action_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_action_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_assoc_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_assoc_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_bind(OM_object \fIsession\fB, OM_workspace \fIworkspace\fB, OM_private_object *\fIbound_session_return\fB);\fR
.HP
\fBMP_status mp_cancel_get_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_cancel_get_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_create_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_create_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_delete_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_objt * \fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_delete_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBOM_sint mp_error_message(MP_status \fIerror\fB, OM_sint \fIlength\fB, unsigned char *\fIerror_text_return\fB);\fR
.HP
\fBMP_status mp_event_report_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_event_report_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_get_assoc_info(OM_private_object \fIreceive_result_or_argument\fB, OM_uint \fIrequest_mask\fB, OM_uint \fIresult_mask\fB, OM_public_object *\fIpres_layer_args\fB, OM_public_object *\fIacse_args\fB, OM_public_object *\fIcmip_assoc_args\fB, OM_public_object *\fIstandard_externals\fB);\fR
.HP
\fBMP_status mp_get_last_error(OM_workspace \fIworkspace\fB, OM_uint32 *\fIadditional_error_return\fB);\fR
.HP
\fBMP_status mp_get_next_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_get_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_get_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBOM_workspace mp_initialize(void);\fR
.HP
\fBMP_status mp_negotiate(MP_feature \fIfeature_list\fB[], OM_workspace \fIworkspace\fB);\fR
.HP
\fBMP_status mp_receive(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_sint *\fIprimitive_return\fB, OM_sint *\fIcompletion_flag_return\fB, MP_status *\fIoperation_notification_status_return\fB, OM_private_object *\fIresult_or_argument_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_release_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_release_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBMP_status mp_set_req(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIargument\fB, OM_private_object *\fIresult_return\fB, OM_sint32 *\fIinvoke_id_return\fB);\fR
.HP
\fBMP_status mp_set_rsp(OM_private_object \fIsession\fB, OM_private_object \fIcontext\fB, OM_object \fIresponse\fB, OM_sint32 \fIinvoke_id\fB);\fR
.HP
\fBvoid mp_shutdown(OM_workspace \fIworkspace\fB);\fR
.HP
\fBMP_status mp_unbind(OM_private_object \fIsession\fB);\fR
.HP
\fBMP_status mp_validate_object(OM_workspace \fIworkspace\fB, OM_object \fItest_object\fB, OM_private_object *\fIbad_argument_return\fB);\fR
.HP
\fBMP_status mp_wait(MP_waiting_sessions \fIbound_session_list\fB[], OM_workspace \fIworkspace\fB, OM_uint32 \fItimeout\fB);\fR
.\"
.\"
.SH FUNCTIONS
.PP
.TP
.BR mp_abandon (3)
abandon locally the result of a pending asynchronous operation.
.TP
.BR mp_abort_req (3)
abort a management association.
.TP
.BR mp_action_req (3)
request an action on managed objects to be performed.
.TP
.BR mp_action_rsp (3)
response to an Action request.
.TP
.BR mp_assoc_req (3)
build a management association.
.TP
.BR mp_assoc_rsp (3)
response to an association request.
.TP
.BR mp_bind (3)
open a managemen session with the Agent(s) or the Manager(s).
.TP
.BR mp_cancel_get_req (3)
cancel an outstanding asynchronous Get operation.
.TP
.BR mp_cancel_get_rsp (3)
response to a Cancel-Get request.
.TP
.BR mp_create_req (3)
request to create an object in the MIB.
.TP
.BR mp_create_rsp (3)
response to a Create request.
.TP
.BR mp_delete_req (3)
request to delete an object in the MIB.
.TP
.BR mp_delete_rsp (3)
response to a Delete request.
.TP
.BR mp_error_message (3)
return an error message describing a particular error.
.TP
.BR mp_event_report_req (3)
request to forward a managment notification.
.TP
.BR mp_event_report_rsp (3)
response to an Event Report request.
.TP
.BR mp_get_assoc_info (3)
return negotiated connection values.
.TP
.BR mp_get_last_error (3)
retrieve secondary error code.
.TP
.BR mp_get_next (3)
request to get the next SNMP management information.
.TP
.BR mp_get_req (3)
request to get management information.
.TP
.BR mp_get_rsp (3)
response to Get request.
.TP
.BR mp_initialize (3)
initialize the interface and create a workspace.
.TP
.BR mp_negotiate (3)
negotiate features of the interface and service.
.TP
.BR mp_receive (3)
retrieve the result of an asynchronously executed management operation or the content of a received
management notification/indication.
.TP
.BR mp_release_req (3)
release a management association.
.TP
.BR mp_release_rsp (3)
response to a Release request.
.TP
.BR mp_set_req (3)
request to modify (replace, add, remove) attribute value(s) of a managed object with the specified
attribute value(s).
.TP
.BR mp_set_rsp (3)
response to a Set request.
.TP
.BR mp_shutdown (3)
discard a workspace.
.TP
.BR mp_unbind (3)
close a management session.
.TP
.BR mp_validate_object (3)
analyze an OM object.
.TP
.BR mp_wait (3)
wait for the availability of management message(s) from one or more bound Sessions.
.\"
.\"
.SH OVERVIEW
.PP
This manual page describes the OpenSS7 Standard XMP Library.
.\"
.\"
.SH DESCRIPTION
.PP
.\"
.SS Negotiation Sequence
.PP
The interface has an initialize and shutdown sequence that permits the
negotiation of optional features.  This involves the functions
.BR mp_initialize (3), 
.BR mp_negotiate (3),
and
.BR mp_shutdown (3).
Every application program must first call
.BR mp_initialize (3),
which returns a workspace.  This workspace supports only the standard Common
Management Service Package.  The workspace can be extended to support either
the CMIS package or the SNMP package or both, packages, or any vendor
extensions.  Vendor extensions may include additional packages, and may also
include additional or modified functionality.  All such packages or other
extensions are identified by means of OSI Object Identifiers, and these Object
Identifiers are supplied to the
.BR mp_negotiate (3)
function to incorporate the extensions into the workspace.  Features defined
by the XMP
.[
xmp c306
.]
specification are described and assigned object identifiers by that package.
A feature represents any package or any additional or modified functionality
that is subject to negotiation.  The
.BR mp_negotiate (3)
function allows some particular features to be made available.
.PP
After a workspace with the required features has been negotiated in this way,
the applications can use the workspace as required.  It can create and
manipulate OM objects using the OM functions,
.[
xom c607
.]
and can start one or more
managemend sessions using
.BR mp_bind (3).
All the sessions on a given workspace share the same features.
.PP
Eventually, when it has completed its tasks, terminating all its management
sessions using
.BR mp_unbind (3),
and released all its OM objects using
.BR om_delete (3),
.[
xom c607
.]
the application should ensure that resources associated with the interface are
freed by calling
.BR mp_shutdown (3).
.PP
A miscellaneous error arises if an attempt is made to use an unavailable
feature.  If an instance of a class that is not in an available package is
supplied as a function argument, the bad-class error arises.
.\"
.\"
.SS Names, Addresses and Titles
.PP
To address a wide variety of management and transport protocols, the interface
is capable of accepting various forms of object names, system addresses and
program or system titles.
.IP \(bu \w'\(bu\(em'u
.B Name
is an \*(lqabstract class\*(rq that contains various subclass types used to
define managed object instances.  For example, the DS-DN subclass is typically
used for ISO mangaed obejct instance naming.
.IP \(bu
.B Address
is an \*(lqabstract class\*(rq that contains varioius subclass types used to
define the specific location to contact a particular agent or manager.  For
example, the Network-Address subclass is typically used to define the location
of an SNMP agent or manager.
.IP \(bu
.B Title
is an \*(lqabstract class\*(rq that contains various subclass types used to
define the specific program or system name responsible for a managed object
instance.
.PP
All three abstract clases participate in an implementation-specific name
resolution scheme.  It is assumed that if given a \fBName\fP, and
implementation can determine the \fBTitle\fP responsible for that \fBName\fP.
It is also assumed that given a \fBTitle\fP, an implementation can determine
the \fBAddress\fP of that \fBTitle\fP.
.PP
Note: The way in which these relationships are resolved is implementation
dependent, but use of the X/Open Directory Services, XDS,
.[
XDS
.]
should play a significant role.
.PP
The performer of an invoked operation or a reported notification may be
explicitly designated at the interface boundary using the following precedence
rules:
.IP 1. \w'0.\(em'u
A default Title and/or Address may be supplied as paramters to a bound
session.  If both are provided, the implementation will verify that the Title
resolves to the Address.
.IP 2.
If automated connection management is used, a performer Title and/or Address
may be supplied as parameters within the context of a specific operation or
notification request.  If both are provided, the implementation will verify
that the Title resolves to the Address.  The context Title and/or Address
takes precedence over the session's Title and/or Address for unconnected
session objects.
.IP 3.
In the case of SNMP, a responder Address may be supplied as a parameter within
the argument of a specific operation or notification request.  The argument
Address takes precedence over either the session Title and/or Address or the
context Title and/or Address.
.IP 4.
If the performer of an invoked operation or a reported notification is not
explicitly designated at the interface boundary, the implementation will
resolve the managed object Name to the appropriate Title and/or Address.
.\"
.\"
.SS Session
.PP
A session identifies to which agent or to which manager a particular
management operation or notification will be sent.  It contains some bind
arguments, such as the name of the requestor.  The session is passed as the
first argument to most interface functions.
.PP
A session is described by an OM object of OM class \fBSession\fP.  It is
created, and appropriate parameter values may be set, using the
OSI-Abstract-Data Manipulation
.[
xom c607
.]
functions.  A management session is then started with
.BR mp_bind (3)
and later is terminated with
.BR mp_unbind (3).
A session with default parameters can be started by passing the constant
\fBOM_DEFAULT_SESSION\fP as the \fBSession\fP argument to
.BR mp_bind (3).
.PP
.BR mp_bind (3)
must be called before the \fBSession\fP can be used as an argument to any
other functions in this instance.  After
.BR mp_unbind (3)
has been called,
.BR mp_bind (3)
must be called again if another session is to be started.
.PP
The interface supports concurrent sessions, so that an application implemented
as a single process, such as a server in a client-server model, can interact
with the Management Information Services using several identities; and so that
a process can interact directly and concurrently with different parts of the
MIB.
.PP
A session can be used either acting as a manager, that is, invoker of
management operations and performer of management notifications, or acting as
an agent, that is, performer of management operations and invoker of
management notifications, or both.
.PP
A session can be restricted for use only with a designated program called the
responder.  When the responder is omitted and automated connection management
is used, the session can be used to exchange management information with all
processes.
.PP
The responder (title and address) parameters of an opened session, if present,
specifies the performer of the requested operation or reported notification.
The precedence reuls on address and title of the responder are described
above.
.PP
Other OM attributes (vendors' implementation extensions) may be included to
specify characteristics of the underlying protocol used.
.PP
There are three types of Session objects:

NEWS
OpenSS7 -- history of user visible changes.  2015-01-15
$Id: NEWS,v 1.1.2.25 2011-05-10 13:45:30 brian Exp $
Copyright (c) 2008-2015  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
'OpenSS7' package.

Major changes for release openss7-1.1.7.20150115
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Source code release on GitHub.

Major changes for release openss7-1.1.7.20131209
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build corrections for RHEL 6.4.

   - Scripts for country boundary, shoreline and rivers database
     preparation.

   - Database for ANSI T1.101 point code assignments.

   - Suppress extransous /proc/1/comm error messages when testing for
     systemd.

   - Handle PUIAS to Springdale rename for PUIAS distribution.

Major changes for release openss7-1.1.7.20131123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Correction to ca-cert handling in RHEL 6 install scriptlet.

   - RHEL 6 build changes.

   - Set permissions correctly when applying kernel module patch.

   - Do not duplicate certs when including ca-cert.

   - Avoid conflicting shelll variable in RPM install scriplet.

   - Remove /proc/1/comm checks for systemd in module loader.

Major changes for release openss7-1.1.7.20130209
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Corrections to RPM spec file.

Major changes for release openss7-1.1.7.20130129
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Compensate for RHEL6 lack of c_rehash.

   - Do not create sysfs nodes: they do not work properly.

   - Use no foreground flag in strace.service file.

   - More error traces for strerr logger.

   - Remove static device names (the do not work either).

   - Changes to support weak kernel modules.

   - Clean up weak module support.

Major changes for release openss7-1.1.7.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Better Archlinux install script support.

Major changes for release openss7-1.1.6.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Remove use of kill_litter_super().

Major changes for release openss7-1.1.3.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Updates and corrections to weak kernel module builds.

   - Repo support and corrections for Archlinux packages.

Major changes for release openss7-1.1.3.20130123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Additional support for Archlinux.

   - Add udev rules for strace and strerr.

   - Add system unit files for strace and strerr.

   - Add repo support for Archlinux packages.

   - Build Archlinux packages better.

Major changes for release openss7-1.1.3.20130121
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build support for SLES 11.

   - Handle certificates better.

   - Get configuration files in the right places for SLES 11.

   - Support in RPM spec file for certificates on SLES 11.

   - Create a new OpenSS7 certificate.

   - Repository support for SLES 11.

   - RPM spec file reworked for new RPM release.

   - Better certificate handling for RHEL.

   - Additional build support for RHEL 6 with certificates.


src/lib/xmp.c
/*****************************************************************************

 @(#) File: src/lib/xmp.c

 -----------------------------------------------------------------------------

 Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com/>
 Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
 Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>

 All Rights Reserved.

 This program is free software: you can redistribute it and/or modify it under
 the terms of the GNU Affero General Public License as published by the Free
 Software Foundation, version 3 of the license.

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
 details.

 You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>, or
 write to the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA
 02139, USA.

 -----------------------------------------------------------------------------

 U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
 behalf of the U.S. Government ("Government"), the following provisions apply
 to you.  If the Software is supplied by the Department of Defense ("DoD"), it
 is classified as "Commercial Computer Software" under paragraph 252.227-7014
 of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
 successor regulations) and the Government is acquiring only the license rights
 granted herein (the license rights customarily provided to non-Government
 users).  If the Software is supplied to any unit or agency of the Government
 other than DoD, it is classified as "Restricted Computer Software" and the
 Government's rights in the Software are defined in paragraph 52.227-19 of the
 Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
 the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
 (or any successor regulations).

 -----------------------------------------------------------------------------

 Commercial licensing and support of this software is available from OpenSS7
 Corporation at a fee.  See http://www.openss7.com/

 *****************************************************************************/

static char const ident[] = "src/lib/xmp.c (" PACKAGE_ENVR ") " PACKAGE_DATE;

#include <xom.h>
#include <xmp.h>

MP_status
mp_abandon(OM_private_object session, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_abandon,mp_abandon@@XMP_1.0");

MP_status
mp_abort_req(OM_private_object session, OM_private_object context, OM_object argument)
{
}

__asm__(".symver __xmp_mp_abort_req,mp_abort_req@@XMP_1.0");

MP_status
mp_action_req(OM_private_object session, OM_private_object context,
	      OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_action_req,mp_action_req@@XMP_1.0");

MP_status
mp_action_rsp(OM_private_object session, OM_private_object context,
	      OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_action_rsp,mp_action_rsp@@XMP_1.0");

MP_status
mp_assoc_req(OM_private_object session, OM_private_object context,
	     OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_assoc_req,mp_assoc_req@@XMP_1.0");

MP_status
mp_assoc_rsp(OM_private_object session, OM_private_object context,
	     OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_assoc_rsp,mp_assoc_rsp@@XMP_1.0");

MP_status
mp_bind(OM_object session, OM_workspace workspace, OM_private_object * bound_session_return)
{
}

__asm__(".symver __xmp_mp_bind,mp_bind@@XMP_1.0");

MP_status
mp_cancel_get_req(OM_private_object session, OM_private_object context,
		  OM_object argument, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_cancel_get_req,mp_cancel_get_req@@XMP_1.0");

MP_status
mp_cancel_get_rsp(OM_private_object session, OM_private_object context,
		  OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_cancel_get_rsp,mp_cancel_get_rsp@@XMP_1.0");

MP_status
mp_create_req(OM_private_object session, OM_private_object context,
	      OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_create_req,mp_create_req@@XMP_1.0");

MP_status
mp_create_rsp(OM_private_object session, OM_private_object context,
	      OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_create_rsp,mp_create_rsp@@XMP_1.0");

MP_status
mp_delete_req(OM_private_object session, OM_private_object context,
	      OM_object argument, OM_private_objt * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_delete_req,mp_delete_req@@XMP_1.0");

MP_status
mp_delete_rsp(OM_private_object session, OM_private_object context,
	      OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_delete_rsp,mp_delete_rsp@@XMP_1.0");

OM_sint
mp_error_message(MP_status error, OM_sint length, unsigned char *error_text_return)
{
}

__asm__(".symver __xmp_mp_error_message,mp_error_message@@XMP_1.0");

MP_status
mp_event_report_req(OM_private_object session, OM_private_object context,
		    OM_object argument, OM_private_object * result_return,
		    OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_event_report_req,mp_event_report_req@@XMP_1.0");

MP_status
mp_event_report_rsp(OM_private_object session, OM_private_object context,
		    OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_event_report_rsp,mp_event_report_rsp@@XMP_1.0");

MP_status
mp_get_assoc_info(OM_private_object receive_result_or_argument,
		  OM_uint request_mask, OM_uint result_mask,
		  OM_public_object * pres_layer_args, OM_public_object * acse_args,
		  OM_public_object * cmip_assoc_args, OM_public_object * standard_externals)
{
}

__asm__(".symver __xmp_mp_get_assoc_info,mp_get_assoc_info@@XMP_1.0");

MP_status
mp_get_last_error(OM_workspace workspace, OM_uint32 * additional_error_return)
{
}

__asm__(".symver __xmp_mp_get_last_error,mp_get_last_error@@XMP_1.0");

MP_status
mp_get_next_req(OM_private_object session, OM_private_object context,
		OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_get_next_req,mp_get_next_req@@XMP_1.0");

MP_status
mp_get_req(OM_private_object session, OM_private_object context,
	   OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_get_req,mp_get_req@@XMP_1.0");

MP_status
mp_get_rsp(OM_private_object session, OM_private_object context,
	   OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_get_rsp,mp_get_rsp@@XMP_1.0");

OM_workspace
mp_initialize(void)
{
}

__asm__(".symver __xmp_mp_initialize,mp_initialize@@XMP_1.0");

MP_status
mp_negotiate(MP_feature feature_list[], OM_workspace workspace)
{
}

__asm__(".symver __xmp_mp_negotiate,mp_negotiate@@XMP_1.0");

MP_status
mp_receive(OM_private_object session, OM_private_object context,
	   OM_sint * primitive_return, OM_sint * completion_flag_return,
	   MP_status * operation_notification_status_return,
	   OM_private_object * result_or_argument_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_receive,mp_receive@@XMP_1.0");

MP_status
mp_release_req(OM_private_object session, OM_private_object context,
	       OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_release_req,mp_release_req@@XMP_1.0");

MP_status
mp_release_rsp(OM_private_object session, OM_private_object context,
	       OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_release_rsp,mp_release_rsp@@XMP_1.0");

MP_status
mp_set_req(OM_private_object session, OM_private_object context,
	   OM_object argument, OM_private_object * result_return, OM_sint32 * invoke_id_return)
{
}

__asm__(".symver __xmp_mp_set_req,mp_set_req@@XMP_1.0");

MP_status
mp_set_rsp(OM_private_object session, OM_private_object context,
	   OM_object response, OM_sint32 invoke_id)
{
}

__asm__(".symver __xmp_mp_set_rsp,mp_set_rsp@@XMP_1.0");

void
mp_shutdown(OM_workspace workspace)
{
}

__asm__(".symver __xmp_mp_shutdown,mp_shutdown@@XMP_1.0");

MP_status
mp_unbind(OM_private_object session)
{
}

__asm__(".symver __xmp_mp_unbind,mp_unbind@@XMP_1.0");

MP_status
mp_validate_object(OM_workspace workspace, OM_object test_object,
		   OM_private_object * bad_argument_return)
{
}

__asm__(".symver __xmp_mp_validate_object,mp_validate_object@@XMP_1.0");

MP_status
mp_wait(MP_waiting_sessions bound_session_list[], OM_workspace workspace, OM_uint32 timeout)
{
}

__asm__(".symver __xmp_mp_wait,mp_wait@@XMP_1.0");

// vim: com=srO\:/**,mb\:*,ex\:*/,srO\:/*,mb\:*,ex\:*/,b\:TRANS
